src/planner/Query.cpp

```diff
@@ -153,7 +154,14 @@ void Project::clone(const Project &p) {
 
 std::unique_ptr<PlanNodeDescription> Project::explain() const {
     auto desc = SingleInputNode::explain();
-    addDescription("columns", cols_ ? cols_->toString() : "", desc.get());
+    if (cols_) {
+        auto columns = folly::dynamic::array();
+        for (const auto* col : cols_->columns()) {
+            DCHECK(col != nullptr);
+            columns.push_back(col->toString());
+        }
+        addDescription("columns", folly::toJson(columns), desc.get());
+    }
```

Comment:
else ?

Reply:
do nothing and don't output columns info

Reply:
Will empty list is more clear?

Reply:
Fixed